### PR TITLE
Add action to replace () with {} in functions and vice versa

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, users would need to manually edit parens to braces if they wanted to include multiple expressions. Now, it's possible to rewrite it automatically.

It's also possible to rewrite braces to parens in case of single statement.

metals-sample-1629574444266

### CODE REVIEW

**Review by dos65 — APPROVED** [review]
LGTM!

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+              codeAction
+          }
+      }
+      .flatten
+      .toSeq
+  }
+}
+
+object RewriteBracesParensCodeAction {
+  val toParens = "Rewrite to parenthesis"
```
Comment (kpodsiad) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala:93]:
```suggestion
  val toParens = "Rewrite to parentheses"
```

**Review by kpodsiad — APPROVED** [review]
Maybe it's worth to add negative test like 
```scala
List(1,2,3).map { <<x>> => 
  println(x)
  x + 2
}
```
?

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala (hunk 1 above)
Comment (ckipp01) [reply metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala:93]: Actually `parenthesis` is correct here.

**Review by ckipp01 — APPROVED** [review]
LGTM!

**Comment by tgodzik** [comment]
> Maybe it's worth to add negative test like
> 
> ```scala
> List(1,2,3).map { <<x>> => 
>   println(x)
>   x + 2
> }
> ```
> 
> ?

Added!

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala (hunk 1 above)
Comment (kpodsiad) [reply metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala:93]: Isn't the `parenthesis` singular?

**Pull request merged**